packages/forklift-console-plugin/src/modules/Providers/views/details/tabs/VirtualMachines/components/VMConcernsCellRenderer.tsx

```diff
@@ -20,7 +20,9 @@ import { VMCellProps } from './VMCellProps';
  * @returns {ReactElement} The rendered table cell.
  */
 export const VMConcernsCellRenderer: React.FC<VMCellProps> = ({ data }) => {
-  const groupedConcerns = groupConcernsByCategory(data?.vm?.concerns);
+  const concerns = data?.vm?.providerType === 'openshift' ? [] : data?.vm?.concerns;
```

Comment:
since this method should not be called with openshit, can we
a. return undefined or <></> in case of openshift
b. raise an error

?

if we are calling this method with openshift we are probably wrong some ware, I prefer to have a 
```
// santiy check
if (openshift) { exit with or without rasing an error }

// continue with the code
``

Reply:
Empty concern array has the benefit of using the same render path through the code as other providers. This makes refactoring safer and easier to maintain.
Next candidate is `<></>` . Raising an exception is risky - we don't have automatic tests so incorrect usage could slip into production and block the functionality.

Reply:
> Empty concern array has the benefit of using the same render path through the code as other providers. This makes refactoring safer and easier to maintain.

Right, i think we have an empty cell for this scenario, so we can return an empty cell, if we don't we can make one for similar cases in the future.

so it will be:
if (openshift) return emptyCell

Reply:
https://github.com/kubev2v/forklift-console-plugin/blob/main/packages/forklift-console-plugin/src/modules/Providers/utils/components/TableCell/TableEmptyCell.tsx#L9